apps/api-gateway/src/api/report_templates.py

```
@@ -0,0 +1,330 @@
+            day_of_month=request.day_of_month,
+        )
+        return sr.to_dict()
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
+
+
+@router.put("/scheduled-reports/{scheduled_id}")
+async def update_scheduled_report(
+    scheduled_id: str,
+    request: ScheduledReportUpdateRequest,
+    current_user: User = Depends(get_current_active_user),
```

Ruff (0.15.2)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)